nautobot/core/management/commands/pre_migrate.py

```diff
@@ -75,6 +118,259 @@ def check_configcontext_uniqueness():
         )
 
 
+def check_permissions_constraints(command):
+    """
+    Check for permission constraints that are referencing an object or field
+    that will be migrated to a different model/field/value in 2.0 migrations.
+
+    Objects that will be replaced in 2.0 are:
+    - dcim.DeviceRole
+    - dcim.RackRole
+    - dcim.Region
+    - dcim.Site
+    - ipam.Aggregate
+    - ipam.Role
+
+    Objects that will possibly have fields migrated to a different value in 2.0 are:
+    - ipam.IPAddress
+    - ipam.Prefix
+    - ipam.Service
+    - extras.CustomField
+    - extras.GitRepository
+    - extras.Job
+    - extras.ObjectPermission
+    - extras.ScheduledJob
+    - extras.TaggedItem
+
+    Objects that had fields renamed or removed in 2.0 are:
+    - circuits.Circuit
+    - ContentType
+    - dcim.CablePath
+    - dcim.Device
+    - dcim.DeviceRedundancyGroup
+    - dcim.DeviceType
+    - dcim.FrontPortTemplate
+    - dcim.InventoryItem
+    - dcim.Location
+    - dcim.PowerOutletTemplate
+    - dcim.PowerPanel
+    - dcim.PowerPort
+    - dcim.PowerPortTemplate
+    - dcim.Rack
+    - dcim.RackGroup
+    - dcim.RearPort
+    - dcim.RearPortTemplate
+    - extras.ComputedField
+    - extras.ConfigContext
+    - extras.ConfigContextSchema
+    - extras.CustomField
+    - extras.CustomFieldChoice
+    - extras.Job
+    - extras.JobResult
+    - extras.Relationship
+    - extras.Secret
+    - extras.SecretsGroup
+    - extras.SecretsGroupAssociation
+    - extras.Status
+    - ipam.IPAddress
+    - ipam.Prefix
+    - ipam.RIR
+    - ipam.Service
+    - ipam.VLAN
+    - ipam.VRF
+    - tenancy.Tenant
+    - tenancy.TenantGroup
+    - users.User
+    - virtualization.Cluster
+    - virtualization.VirtualMachine
+    """
+
+    replaced_models = [
+        Aggregate,
+        DeviceRole,
+        RackRole,
+        Region,
+        Role,
+        Site,
+    ]
+
+    pk_change_models = [
+        IPAddress,
+        Prefix,
```

Comment:
The migrations aren't changing the PKs or doing any deduplication but I still think it's a good idea to bring any constraints using those models' PKs to the users' attention because their prefixes and ip addresses are likely going to need to be deduplicated manually after the migration and the instance that they choose to keep may not match the instance that they originally pointed the constraint to. It's probably highly unlikely that a constraint is pointing to one instance of a duplicate IP/Prefix so maybe it's ok to leave it out. I don't think it hurts to leave it in either though?

Reply:
Makes sense, thanks for the context. I worry that the output of this command will already be potentially overwhelming for users with any realistically complex set of defined permissions, but it's probably better to err on the side of a few false-positive warnings than potentially miss important changes.